packages/amplify-graphql-transformer-core/src/transformer-context/output.ts

```diff
@@ -297,13 +344,17 @@ export class TransformerOutput implements TransformerContextOutputProvider {
    * @param fields The fields to add the subscription type.
    */
   public addSubscriptionFields(fields: FieldDefinitionNode[]) {
+    this.ensureSubscription();
     const subscriptionTypeName = this.getSubscriptionTypeName();
     if (subscriptionTypeName) {
       if (!this.getType(subscriptionTypeName)) {
         this.addType(blankObject(subscriptionTypeName));
       }
       let subscriptionType = objectExtension(subscriptionTypeName, fields);
       this.addObjectExtension(subscriptionType);
+    } else {
+      this.addDefaultSubscription();
```

Comment:
This should be covered by the call to `this.ensureSubscription()`, right?

Reply:
Oops, yes. This is the original implementation before I changed it use `ensureSubscription`.